### PR TITLE
HPCC-18553 Non-posix DropZone paths should be translated by the environment.

### PR DESCRIPTION
**Pull request by AttilaVamos**
Use getStandardPosixPath(StringBuffer &result, const char *path) function to
translate drop zone path to posix format before store it into the local cache
    
Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:

Tested manually with non posix drop zone path in environment.xml and without a real non posix drop zone.

### CODE REVIEW

**Comment by AttilaVamos** [comment]
@jakesmith please review my changes.

File: common/environment/environment.cpp [hunk 1]
```diff
@@ -922,7 +922,12 @@ class CConstDropZoneInfo : public CConstEnvBase, implements IConstDropZoneInfo
 public:
     IMPLEMENT_IINTERFACE;
     IMPLEMENT_ICONSTENVBASE;
-    CConstDropZoneInfo(CLocalEnvironment *env, IPropertyTree *root) : CConstEnvBase(env, root) {}
+    CConstDropZoneInfo(CLocalEnvironment *env, IPropertyTree *root) : CConstEnvBase(env, root)
+    {
+        StringBuffer posixPath;
+        getStandardPosixPath(posixPath, root->queryProp("@directory"));
+        root->setProp("@directory", posixPath.str());
```
Comment (jakesmith) [inline common/environment/environment.cpp]: You should not call setX on the trees.
If there tree is from Dali - you will be altering the contents there.

**Comment by jakesmith** [comment]
@AttilaVamos - 1 comment.

**Comment by jakesmith** [comment]
@AttilaVamos - have you verified it translates as expected ?

**Comment by AttilaVamos** [comment]
@jakesmith yes, I do. I created an environment.xml with one posix and one nonposix drop zone path, then I try to spray a file to posix one (because it is a real drop zone). During the spray I traced the execution and the nonposix patch converted to posix form when the local environment cache build.

**Comment by jakesmith** [comment]
@AttilaVamos - comment needs addressing.

File: common/environment/environment.cpp (hunk 1 above)
Comment (AttilaVamos) [reply common/environment/environment.cpp]: Changed

**Comment by AttilaVamos** [comment]
@jakesmith comment addressed.

**Comment by jakesmith** [comment]
@AttilaVamos - looks fine.

Need to check some tick boxes above though

**Comment by jakesmith** [comment]
Looks good.
This PR depends on Kevin's PR being merged first (and then this will need rebasing).
So can't be merged yet.

**Comment by richardkchapman** [comment]
Kevin's has been merged - @AttilaVamos can you rebase please

**Comment by AttilaVamos** [comment]
@richardkchapman rebased.

**Pull request merged**